src/libraries/System.Formats.Tar/src/System/Formats/Tar/PaxTarEntry.cs

```diff
@@ -119,5 +150,29 @@ public PaxTarEntry(TarEntryType entryType, string entryName, IEnumerable<KeyValu
 
         // Determines if the current instance's entry type supports setting a data stream.
         internal override bool IsDataStreamSetterSupported() => EntryType == TarEntryType.RegularFile;
+
+        // Checks if the extended attributes dictionary contains 'atime' and 'ctime'. If any of them is not found, it is added with the value of either the current entry's 'mtime', or 'DateTimeOffset.UtcNow', depending on the value of 'useMTime'.
+        private void AddNewAccessAndChangeTimestampsIfNotExist(bool useMTime)
+        {
+            Debug.Assert(_header._extendedAttributes != null);
+
+            bool containsATime = _header._extendedAttributes.ContainsKey(TarHeader.PaxEaATime);
+            bool containsCTime = _header._extendedAttributes.ContainsKey(TarHeader.PaxEaCTime);
+
+            if (!containsATime || !containsCTime)
+            {
+                string secondsFromEpochString = TarHelpers.GetTimestampStringFromDateTimeOffset(useMTime ? _header._mTime : DateTimeOffset.UtcNow);
```

Comment:
Is there a possibility that useMTime is true but _header._mTime is uninitialized?

Reply:
It shouldn't. The `base` constructor that is called when the `useMTime` argument is true, is the constructor in `TarEntry` that sets the value of `_mTime` to `DateTime.UtcNow`.
I added a couple of debug asserts to ensure we have non-default `mTime` here.